pkg/preemptiontoleration/preemption_toleration.go

```diff
@@ -203,18 +292,19 @@ func (pl *PreemptionToleration) FindCandidates(ctx context.Context, state *frame
 // returns preemption candidates and a map indicating filtered nodes statuses.
 // The number of candidates depends on the constraints defined in the plugin's args. In the returned list of
 // candidates, ones that do not violate PDB are preferred over ones that do.
-func dryRunPreemption(ctx context.Context, fh framework.Handle,
+func (pl *PreemptionToleration) dryRunPreemption(ctx context.Context, fh framework.Handle,
 	state *framework.CycleState, pod *v1.Pod, potentialNodes []*framework.NodeInfo,
 	pdbs []*policy.PodDisruptionBudget, offset int32, numCandidates int32) ([]dp.Candidate, framework.NodeToStatusMap) {
 	nonViolatingCandidates := newCandidateList(numCandidates)
 	violatingCandidates := newCandidateList(numCandidates)
 	parallelCtx, cancel := context.WithCancel(ctx)
 	nodeStatuses := make(framework.NodeToStatusMap)
 	var statusesLock sync.Mutex
+	now := pl.clock.Now()
 	checkNode := func(i int) {
 		nodeInfoCopy := potentialNodes[(int(offset)+i)%len(potentialNodes)].Clone()
 		stateCopy := state.Clone()
-		pods, numPDBViolations, status := selectVictimsOnNode(ctx, fh, stateCopy, pod, nodeInfoCopy, pdbs)
+		pods, numPDBViolations, status := pl.selectVictimsOnNode(ctx, fh, stateCopy, pod, nodeInfoCopy, pdbs, now)
```

Comment:
> Cannot we just use pl.clock in selectVictimsOnNode()?

I would like to evaluate all the preemption toleration policies with the same timestamp of `now` because this evaluates toleration seconds from pods' scheduled timestamps.  It is because `CanToleratePlugin` function which is expected to be public (addressed in TODO) does have `now` as its argument. So, passing `now` to `selectVictimsOnNode` is just to wire in `CanToleratePlugin` function.